src/commands.cpp

```diff
@@ -8261,4 +8261,114 @@ bool CommandBackupRemove::procresult(Result r)
     return r.wasErrorOrOK();
 }
 
+CommandGetBanners::CommandGetBanners(MegaClient* client)
+{
+    cmd("gban");
+
+    tag = client->reqtag;
+}
+
+bool CommandGetBanners::procresult(Result r)
+{
+    if (r.wasErrorOrOK())
+    {
+        client->app->getbanner_result(r.errorOrOK());
+        return true; // because parsing didn't fail
+    }
+
+    /*
+    [
+        [
+            {
+                "id": 2, ///The banner id
+                "t": "R2V0IFZlcmlmaWVk", ///Banner title
+                "d": "TWFrZSBpdCBlYXNpZXIgZm9yIHlvdXIgY29udGFjdHMgdG8gZmluZCB5b3Ugb24gTUVHQS4", ///Banner description.
+                "img": "Verified_image.png", ///Image name.
+                "l": "", ///URL
+                "bimg": "Verified_BG.png", ///background image name.
+                "dsp": "https://web-sandbox3.developers.mega.co.nz/images/mega/" ///Where to get the image.
+            },
+            {
+                ...
+            }
+        ]
+    ]
+    */
+
+    if (!client->json.enterarray() ||
+        !client->json.enterarray())
+    {
+        client->app->getbanner_result(API_EARGS);
+        return false; // parsing failed
+    }
+
+    vector< tuple<int, string, string, string, string, string, string> > banners;
+
+    // loop array elements
+    while (client->json.enterobject())
+    {
+        int fieldcount = 7,
+            id = 0;
+        string title, description, img, url, bimg, dsp;
+        bool ok = true;
+
+        // loop and read object members
+        for (int i = 0; ok && i < fieldcount; ++i)
+        {
+            switch (client->json.getnameid())
+            {
+            case MAKENAMEID2('i', 'd'):
+                id = client->json.getint32();
+                ok = id != -1;
+                break;
+
+            case MAKENAMEID1('t'):
+                ok = client->json.storeobject(&title);
+                break;
+
+            case MAKENAMEID1('d'):
+                ok = client->json.storeobject(&description);
+                break;
+
+            case MAKENAMEID3('i', 'm', 'g'):
+                ok = client->json.storeobject(&img);
+                break;
+
+            case MAKENAMEID1('l'):
+                ok = client->json.storeobject(&url);
+                break;
+
+            case MAKENAMEID4('b', 'i', 'm', 'g'):
+                ok = client->json.storeobject(&bimg);
+                break;
+
+            case MAKENAMEID3('d', 's', 'p'):
+                ok = client->json.storeobject(&dsp);
+                break;
+
+            case EOO:  // [[fallthrough]];
+            default:
```

Comment:
We need to be prepared for future changes in the API, where they can add new fields. The parser should skip the unknown items, in that case, so the `default` should consume that part of the JSON (see other commands as reference, you'll see a `storeobject()` in there). In consequence, the `EOO` should be the exit path of the loop (you can use a `bool exit = true` for that purpose).
On the other hand, the command should usually attempt to read the JSON response to the command in full. It means that, in case one of the items cannot be read successfully, we may want to keep going (instead of breaking the loop due to `!ok`). Sometimes, an issue parsing one of the items is not a critical issue. The sanity checks should be done upon `EOO`, where some tests are done against the received values to double-check they are valid and make sense.

Reply:
Updated.
I can't think of some good sanity checks now, but I can add some upon suggestions.